compiler/resolution/functionResolution.cpp

```diff
@@ -8265,7 +8271,17 @@ postFold(Expr* expr) {
       }
 
       if (sym->symbol()->type != dtUnknown && sym->symbol()->type != val->type) {
-        CallExpr* cast = createCast(val, sym->symbol());
+        Symbol* toSym = sym->symbol();
+        if (!toSym->hasFlag(FLAG_TYPE_VARIABLE)) {
+          // This assertion is here to ensure compatibility with old code.
+          // Todo: remove after 1.15 release.
+          Symbol* newToSym = toSym->type->symbol;
+          INT_ASSERT(toSym->type       // cast machinery used this previously
+                  == newToSym->type);  // now it will use this
```

Comment:
I'm wondering if you mean for this to be an WARNING of some sort, given that the next line wouldn't actually ever run with INT_ASSERT?